File: 238-product-of-array-except-self/238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res = []
        l, r = [1]*len(nums), [1]*len(nums)
        for i in range(1, len(nums)):
            l[i] = l[i-1] * nums[i-1]
        
        for i in range(len(nums)-2,-1,-1):
            r[i] = r[i+1] * nums[i+1]
        
        for i in range(len(nums)):
            res.append(l[i]*r[i])
        return res

File: 238-product-of-array-except-self/238-product-of-array-except-self.go
func productExceptSelf(nums []int) []int {
    x := make([]int, len(nums))
    
    x[0] = 1
    
    for i:=1;i<len(nums);i++{
        x[i] = x[i-1] * nums[i-1]
    }
    
    temp := nums[len(nums)-1]
    
    for i:=len(nums)-2;i>=0;i--{
        x[i] *= temp
        temp *= nums[i]
    }
    
    return x
}


